### PR TITLE
[stable/kube-registry-proxy] Update api versions to support k8s 1.16

### PR DESCRIPTION
**Pull request by antongorkovenko**
Signed-off-by: Anton Gorkovenko <gostom@gmail.com>

#### Is this a new chart
no
#### What this PR does / why we need it:
Update api versions to support k8s 1.16
#### Which issue this PR fixes
no
#### Special notes for your reviewer:
no
#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: incubator/kube-registry-proxy/templates/daemon.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ metadata:
     release: {{ .Release.Name | quote }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
 spec:
+  selector:
+    matchLabels:
+      release: {{ .Release.Name | quote }}
```
Comment (zanhsieh) [inline incubator/kube-registry-proxy/templates/daemon.yaml:14]: @antongorkovenko 

I1016 22:10:27.103] Error: release kube-registry-proxy-u2no0fg1nr failed: DaemonSet.apps "kube-registry-proxy-u2no0fg1nr-kube-registry-proxy" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"app":"kube-registry-proxy-u2no0fg1nr-kube-registry-proxy"}: `selector` does not match template `labels`

Comment (antongorkovenko) [reply incubator/kube-registry-proxy/templates/daemon.yaml:14]: it was caused by inconsistency between main and template labels
fixed

**Comment by antongorkovenko** [comment]
@mattfarina @unguiculus @maorfr  please review

**Comment by maorfr** [comment]
i would prefer if we retain backwards compatibility. #17268 has an example. wdyt?

**Comment by antongorkovenko** [comment]
> i would prefer if we retain backwards compatibility. #17268 has an example. wdyt?

I don't like it. New api version are supported in k8s 1.10+. Using of k8s <1.10 is strongly not recommended (deprecated and no support in community and vendor k8s)

**Comment by antongorkovenko** [comment]
@maorfr any updates?

**Comment by antongorkovenko** [comment]
@maorfr

**Pull request closed without merging**